src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -1041,5 +1053,16 @@ public void EmitMetadata(ExpressionEmitter emitter, ImmutableArray<DeclaredMetad
                 }
             });
         }
+
+        private static IEnumerable<string> GetEnabledExperimentalArmFeatures(IFeatureProvider features)
+        {
+            foreach (var (enabled, name) in new[] { (features.ExtensibilityEnabled, "Extensibility"), (features.AssertsEnabled, "Asserts") })
+            {
+                if (enabled)
+                {
+                    yield return name;
+                }
+            }
+        }
```

Comment:
Worth combining this with the `ExperimentalFeatureWarningProvider` you've added?

Reply:
I combined this with `ExperimentalFeatureWarningProvider` logic into a method on `IFeatureProvider` that returns a tuple of (name, <should warn in CLI>, <should inject warning into ARM JSON template>)